### PR TITLE
documenting the legend positions and style possibilities

### PR DESCRIPTION
**Pull request by flothesof**
I just hit the "undocumented legend features" problem and saw that #3505 discussed accepting PRs for documenting these features. So I came up with this PR that extends the customizing plots notebook with some notes about legends and how they can be moved around and styled, as suggested by @philippjfr in the issue.

Let me know what you think.

Cheers
Florian

### CODE REVIEW

**Comment by philippjfr** [comment]
Thanks @flothesof, much appreciated!

File: examples/user_guide/Customizing_Plots.ipynb [hunk 1]
```diff
@@ -154,6 +154,84 @@
...
+    "### Applying customizations to legends\n",
+    "\n",
+    "When overlaying plots with different labels, a legend automatically appears to differentiate elements in the overlay. This legend can be customized in several ways:\n",
+    "\n",
+    "- by **position**\n",
+    "    - by adjusting the legend location within the figure using the `legend_position` option (e.g. `legend_position='bottom_right'`)\n",
+    "    - by adjusting the legend location *outside* of the figure using the `legend_position` and `legend_offset` parameters (which then positions the legend in *screen* space) (e.g. `legend_position='right', legend_offset=(0, 200)`)\n",
+    "- by **style**\n",
+    "    - by muting elements with `legend_muted=True`\n",
+    "    - by putting the legend elements in a column layout with `legend_cols=True`\n",
```
Comment (philippjfr) [inline examples/user_guide/Customizing_Plots.ipynb:170]: Maybe note which of these options are specific to bokeh. I believe legend_position is supported by matplotlib as well and `legend_cols` even takes an integer in matplotlib. The other options are not available in bokeh.

Comment (flothesof) [reply examples/user_guide/Customizing_Plots.ipynb]: I've tried all examples with matplotlib and I came to the conclusion legend_muted and legend_cols only works with bokeh.
Also legend_offset is bokeh_only too.
I've added some texts to reflect this.

Comment (philippjfr) [reply examples/user_guide/Customizing_Plots.ipynb]: IIRC `legend_cols` works as an integer in matplotlib, i.e. it actually specifies the number of columns, something I hope to implement for bokeh eventually.

**Comment by philippjfr** [comment]
I'll merge and fix up the legend_cols docs separately, thanks again!

**Pull request merged**